gcp_deepvariant_runner.py

```diff
@@ -166,6 +168,62 @@
 }}
 """
 
+# Following const values are used to automatically set the computational flags.
+_WGS_MODEL_IDENTIFIER = 'wgs_standard'
+_WES_MODEL_IDENTIFIER = 'wes_standard'
+_WES_LARGE_THR = 12 * 1024 * 1024 * 1024
+_WGS_LARGE_THR = 200 * 1024 * 1024 * 1024
+_WGS_SMALL_THR = 25 * 1024 * 1024 * 1024
+
+
+class BamCategories(enum.Enum):
+  """List of BAM categories that determine automatically assigned flags."""
+  WES_SMALL = 0
+  WES_LARGE = 1
+  WGS_SMALL = 2
+  WGS_MEDIUM = 3
+  WGS_LARGE = 4
+
+
+# Default optimal computational flag values, one per BAM category.
+_DEFAULT_FLAGS = {}
+_DEFAULT_FLAGS[BamCategories.WES_SMALL] = {
+    'make_examples_workers': 8,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 1,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WES_LARGE] = {
+    'make_examples_workers': 16,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 2,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_SMALL] = {
+    'make_examples_workers': 32,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 4,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_MEDIUM] = {
+    'make_examples_workers': 64,
+    'make_examples_cores_per_worker': 1,
+    'tpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_LARGE] = {
+    'make_examples_workers': 128,
+    'make_examples_cores_per_worker': 1,
+    'tpu': True
+}
+# Common computational flag values across all BAM categories.
+_RAM_PER_CORE = 4
```

Comment:
Should we make it variable and put its value per config into _DEFAULT_FLAGS instead?
I recall a fix value of 4 had a side of effect of choosing custom machine that costs us ~20cent(?) more per run, while I am not sure 4GB per core is what needed.

Reply:
The cost difference is 4GB vs 3.75GB memory per core, or $0.000223 per core/hour which is negligible.

For example, a 8 core standard preemptible machine costs $0.0800 per hour, while our custom machine costs 8*($0.006655 + 4*$0.000892)=$0.081784. Difference here is $0.001784, less than 0.18cents per hour. In other words, our custom 8 core machine after 6 hours costs 1 cent more then standard 8 cores machine.

I think this cost is negligible, specially that it will make code cleaner so that we will only deal with # workers and # cores.